src/commons/crypto/signing/dispatch/signerprovider.rs

```diff
@@ -15,47 +15,140 @@ use crate::commons::{
 
 //------------ SignerProvider ------------------------------------------------
 
+#[cfg(not(feature = "hsm"))]
+#[derive(Debug)]
+pub(crate) struct SignerFlags;
+
+#[cfg(not(feature = "hsm"))]
+impl Default for SignerFlags {
+    fn default() -> Self {
+        Self
+    }
+}
+
+#[cfg(feature = "hsm")]
```

Comment:
I get the feeling that the 'hsm' feature flag could be used less. I.e. let limit the available options for the `SignerProvider` enum, but essentially have the same boilerplate always. This is not urgent, we can  look at this later.. also once all of this is merged and has had more field testing we may even just want to remove the feature as such an leave it to config to decide to use hsms or not.